app/src/main/java/com/example/cardgame/GameplayFragment.kt

```diff
@@ -0,0 +1,380 @@
+                }
+            }
+        })
+    }
+
+    //toDo I'll probably delete this method
+    private fun callApiForNextCard(root: View): Int {
+        val client = OkHttpClient()
+        val deckApi = "https://deckofcardsapi.com/api/deck/$deckId/draw/?count=1"
+        val request = Request.Builder().url(deckApi).build()
+
+        client.newCall(request).enqueue(object : Callback {
+            override fun onFailure(call: Call, e: IOException) {
+                e.printStackTrace()
+                Log.d(TAG, "REQUEST FOR NEW CARD FAILED!", e)
+            }
+
+            override fun onResponse(call: Call, response: Response) {
+                if(response.isSuccessful) {
+                    val result = response.body()!!.string()
+                    val strRes = Gson().fromJson<ResponseForNewCard>(result,
+                        ResponseForNewCard::class.java)
+                    val image = strRes.cards[0].image
+                    val cardValue = strRes.cards[0].value
+                    returnCard = Consts.indexOf(cardValue)
+                    activity?.runOnUiThread {
+                        val imgView = root.findViewById<ImageView>(R.id.ivCurrentCard)
+                        Picasso.get().load(image).into(imgView)
+                    }
+                }
+                else {
+                    Log.d(TAG, "RESPONSE FOR NEW CARD IS UNSUCCESSFUL")
+                }
+            }
+        })
+        return returnCard
+    }
+
+    fun endGame() {
+        activity?.getPreferences(Context.MODE_PRIVATE)?.let {
+            it.edit().putInt(Consts.SAVED_CARD_KEY, -1).commit()
+            it.edit().putInt(Consts.SAVED_SCORE_KEY, 0).commit()
+            Log.d(TAG, "sharedPrefs deleted from endGame")
+        }
+        theGameHasEnded = true
+        val bundle = Bundle()
+        bundle.putInt(GAME_SCORE_KEY, currentScore)
+
+        activity?.supportFragmentManager?.popBackStack()
+        //toDo
+        // Save the score if it's in top 10
+    }
+
+    override fun onPause() {
+        super.onPause()
+        if(!theGameHasEnded) {
+            activity?.getPreferences(Context.MODE_PRIVATE)?.let {
+                with (it.edit()) {
+                    putInt(Consts.SAVED_SCORE_KEY, currentScore)
+                    putInt(Consts.SAVED_CARD_KEY, currentCard)
+                    putString(Consts.SAVED_DECK_ID, deckId)
+                    putString(Consts.SAVED_IMAGE_KEY, currentImage)
+                    commit()
+                }
+            }
+        }
+        else {
+            activity?.getPreferences(Context.MODE_PRIVATE)?.let {
+                it.edit().clear().commit()
+                Log.d(TAG, "sharedPrefs deleted from endGame")
+            }
+        }
+
+        Log.d(TAG, "onPause")
+    }
+
+
+    //toDo Put these classes in a separate folder
+    data class ResponseNewDeck (
+        @field:SerializedName("success")
```

Comment:
You dont need `@field:SerializedName` if the serialized name matches the val name (perks of using Kotlin over Java)